pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -167,6 +167,9 @@ func (c *SyncController) CreateResourceQuota(ns *corev1.Namespace) (runtime.Obje
 		}
 		operationErr = c.updateResourceQuota(existing, newQuotaSpec)
 	case "delete":
+		// To allow annotation deletion.
+		namespaceutil.SetNamespaceCondition(ns, time.Second*1, ResourceQuotaInitCondition, false, "")
```

Comment:
It's a little hard to read what this is doing since at first glance the ResourceQuotaInitCondition doesn't have anything to do with deleting and this condition won't ever actually be set on the object in etcd. It would be clearer in my opinion to just go ahead and update the namespace object here, the same way e.g. `validateAndSetNamespaceQuota` calls update. That might be just a personal preference though.

Reply:
Good point, adjusted!